.github/workflows/contract-tests.yml

```diff
@@ -0,0 +1,38 @@
+name: Contract test
+
+env:
+  PACT_USER: ${{ secrets.PACT_USER }}
+  PACT_PASSWORD: ${{ secrets.PACT_PASSWORD }}
+  PACT_URL: ${{ secrets.PACT_URL }}
+  GIT_BRANCH: ${{ github.head_ref }}
+  CONSUMER_APP_VERSION: ${{ github.event.pull_request.head.sha }}
+  PACT_BROKER_SOURCE_HEADER: ${{ secrets.PACT_BROKER_SOURCE_HEADER }}
+
+on:
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - ready_for_review
+      - synchronize
+  merge_group:
+    # merge main
+    branches: [ main ]
+
+jobs:
+  contract-testing:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v3
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: 'gradle'
+      - name: Run contract tests
+        run: ./gradlew --parallel ipv-api:contractTest
```

Comment:
Happy for this as this Spike was specific for IPV but as we add more contract tests for more coverage do we need to change this to '*:contractTest' ?

Reply:
Yeah, will change when relevant